File: Claude/Intermediárias/two_sum/two_sum.cpp
#include <iostream>
#include <stdexcept>
#include <cassert>

unsigned long long factorial(int n) {
    if (n < 0) {
        throw std::invalid_argument("O fatorial não está definido para números negativos");
    }
    if (n == 0 || n == 1) {
        return 1;
    }
    return n * factorial(n - 1);
}

void runTests() {
    assert(factorial(0) == 1);
    assert(factorial(1) == 1);
    assert(factorial(5) == 120);

    try {
        factorial(-1);
        assert(false); // Deve lançar uma exceção
    } catch (const std::invalid_argument& e) {
        // Exceção esperada
    }

    std::cout << "Todos os testes passaram!" << std::endl;
}

int main() {
    runTests();
    return 0;
}

File: Claude/Intermediárias/two_sum/two_sum.java
public class Factorial {

    public static long factorial(int n) {
        if (n < 0) {
            throw new IllegalArgumentException(
                "O fatorial não está definido para números negativos"
            );
        }
        if (n == 0 || n == 1) {
            return 1;
        }
        return n * factorial(n - 1);
    }

    public static void main(String[] args) {
        runTests();
    }

    public static void runTests() {
        assert factorial(0) == 1 : "Teste falhou para factorial(0)";
        assert factorial(1) == 1 : "Teste falhou para factorial(1)";
        assert factorial(5) == 120 : "Teste falhou para factorial(5)";

        try {
            factorial(-1);
            assert false : "Deveria ter lançado uma exceção para entrada negativa";
        } catch (IllegalArgumentException e) {
            // Exceção esperada
        }

        System.out.println("Todos os testes passaram!");
    }
}
